### ISSUE TITLE
Change a textfield for VoiceOver/TalkBack when it is wrapped in Semantics()

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#45046]
b/144562373 internally

When extra information is needed to be added to a text field we could do this:

```
Semantics(
  label: 'Phone Number',
  child: TextField(),
);
```

However we cannot format the text field for semantics e.g. adding a space in between to force talkback/voice over to read letter by letter.

From @goderbauer 
Adding the label as a suffix seems like a bad idea because that's not how text fields work natively. You should provide label ("what should be entered in the textfield", e.g. "phone number") and value ("the current content of the textfield", e.g. 748-328-8272) for the textfield separately and let the platform figure out how and in what order to read them.

Changing how the value of a textfield is read out with VoiceOver/TalkBack seems like a valid feature request. Feel free to file an issue for it and somebody on the team can look into it if that's feasible.

Marking this as critical as the customer is currently resorting to platform specific hacks.

**Comment by goderbauer** [comment]
/cc @darrenaustin 

Thinking about this more I am actually not sure that Android and iOS allow us to specify a different string for the announcement. It is possible that the announcement string and the string in the textfield have to be the same per platform limitations.

**Comment by Hixie** [comment]
@darrenaustin please set a conservative target milestone, thanks.

**Comment by Hixie** [comment]
We don't want to support this, because we think it is the wrong thing for accessibility users. We are not aware of any affordance in the various operating systems a11y APIs that allow this kind of behaviour.

If you wish to ignore our advice and ignore the a11y tools, you could force this behaviour by overridding the relevant Flutter classes and replacing the string in the data we send to the OS. We believe that would likely cause any number of additional accessibility problems for users and would strongly recommend against it.

Please don't hesitate to reopen this bug if you are aware of an OS a11y API that would achieve the effect desired here and think we should implement that API.
